### DISCUSSION TITLE
What is a librarian?

### DISCUSSION THREAD

**Discussion by jeffythedragonslayer** [cc65/cc65#1553] [Q&A] [answered]
Documentation says that this project includes a "librarian."  What is a librarian?

**Comment by mrdudz** [comment, marked as answer]
Am Mittwoch, 25. August 2021, 04:47:26 CEST schrieb Jeff Linahan:
> Documentation says that this project includes a "librarian."  What is a
> librarian?

the assembler produces object files that can be linked (=combined) to produce
the final binary.

a library is simply a collection ("archive") of several such object files, so
you can tell the linker to link all object files from a library that are being
referenced.

a librarian is the program that combines object files to form a library.

--

http://hitmen.eu                 http://ar.pokefinder.org
http://vice-emu.sourceforge.net  http://magicdisk.untergrund.net

Der Drache lehrt: Wer hoch steigen will, muss es gegen den Wind tun.
<Chinesische Weisheit>

**Reply by dakk** [reply]
> Am Mittwoch, 25. August 2021, 04:47:26 CEST schrieb Jeff Linahan:
> a librarian is the program that combines object files to form a library.

I think that the correct name for that software is "linker"

**Reply by mrdudz** [reply]
> I think that the correct name for that software is "linker"

no. The linker combines object files into the final binary.

assembler -> object files -> librarian -> library

object files and libraries -> linker -> binary

**Reply by AshtonSnapp** [reply]
Calling it an archiver instead of a librarian might be less confusing, since that's basically what it is.

**Comment by polluks** [comment]
@jeffythedragonslayer Our librarian is called ar65.

**Reply by groessler** [reply]
Unix people call it "archiver" (ar), MSDOS/Windows/OS2 people call it "librarian" (lib.exe). It's the same thing.

**Comment by groessler** [comment]
I guess that's answered now?

**Reply by jeffythedragonslayer** [reply]
Yes thank you
